### DISCUSSION TITLE
Any way to make a 20x4 LCD display work?

### DISCUSSION THREAD

**Discussion by mzierer** [probonopd/MiniDexed#257] [Q&A]
I'm running Minidexed on a Raspberry 1 and it works like a charm. Sure, no effects and only one TG, but that's enough for me. I2C audio and the encoder work perfectly. What doesn't work though is my display, which is an older Sparkfun, 20x4 HD44780 based LCD I had lying around.

The LCD unfortunately has no I2C connectivity (only UART), so I connected it via the parallel pins (RS, E, DB04...). I double and triple checked all the pins and the corresponding GPIOs, but instead of the correct text, all I get is one blinking black dot that sometimes upredictably changes place when I blindly change menus via the encoder.

Since the protocol is the same as for 16x2 LCDs, I assumed it might just work, but unfortunately it doesn't. I saw that in the code there's a place where you can change the number of rows and lines for the LCD. Could it be that easy? Any other idea how to make this work? What would be the expected behaviour when connecting a 20x4 LCD as is?

**Comment by diyelectromusic** [comment]
I don't know if it is related to your problem or not, but the userinterface code is currently quite dependent on screen width, so when I was playing around with ssd1306 displays, I had to leave a hard-coded 16-character wide value for use with the volume/panning part of the UI whilst allowing everything else to support a configurable 20-character display... so it isn't quite as trivial as it appears! :)

See: https://github.com/diyelectromusic/MiniDexed/tree/ssd1306/src

Kevin

**Reply by probonopd** [reply]
Correct. Similar discussion: https://github.com/probonopd/MiniDexed/discussions/251
